### DISCUSSION TITLE
Host Website of Github with local copy on server

### DISCUSSION THREAD

**Discussion by niclasreich** [community/community#21955] [Other Feature Feedback, Questions, & Ideas]
<p>Good evening!</p>
<p>Long ago - during some downtime of my server - I had the idea to host my (mainly static) website on GitHub. Unfortunately, custom domains do not work with HTTPS/SSL, which is the decisive fact not to do it.</p>
<p>Anyways, I’d like to store my content mainly on my Laptop. I use the GitHub program and every time I’d change something I would push it to my project, right? And then I would like to “pull” (?) it to my server. Is this possible?</p>

**Comment by nadiajoyce** [comment]
<p>Hi <a class="mention" href="/niclasreich">@niclasreich</a>,</p>
<p>This post was moved to a different board that fits your topic of discussion a bit better. This means you’ll get better engagement on your post, and it keeps our Community organized so users can more easily find information.</p>
<p>As you’ll notice, your Topic is now in the Pages board. You can continue the conversation as normal here.</p>

**Comment by birdcar** [comment, marked as answer]
<blockquote>
<p><a class="mention" href="/niclasreich">@niclasreich</a> wrote:</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>Anyways, I’d like to store my content mainly on my Laptop. I use the GitHub program and every time I’d change something I would push it to my project, right? And then I would like to “pull” (?) it to my server. Is this possible?</p>
</blockquote>
<p>It sounds like what you’d like to do is have the code from your repository deploy to your server, is that correct? </p>
<p>If so, this is definitely possible, but requires some configuration. We have a guide that describes how to build a server that uses the <a href="https://developer.github.com/v3/repos/deployments/" rel="nofollow noopener">Deployments API</a> to easily get your code from GitHub into production here:</p>
<p><a href="https://developer.github.com/v3/guides/delivering-deployments/" rel="nofollow noopener">https://developer.github.com/v3/guides/delivering-deployments/</a></p>
<p>Alternatively, you could use the GitHub Auto-Deployment service to receive changes made to your repository and configure it to deliver a deployment to integrators:</p>
<p><a href="https://developer.github.com/v3/guides/automating-deployments-to-integrators/" rel="nofollow noopener">https://developer.github.com/v3/guides/automating-deployments-to-integrators/</a></p>
